### PR TITLE
add optional overridden httplink

### PR DESCRIPTION
**Pull request by hungnguyen-ailo**
- Cubejs's graphql isn't apollo so it doesn't accept batched queries, it just throws a 400 error `Message: Must provide query string., Location: undefined, Path: undefined`. So I added an optional `overriddenHttpLink` for the GraphQLClient to accept an alternative HttpLink.
- It can also be a flag, like `useSingularHttpLink` and the constructor figures out itself

### CODE REVIEW

File: src/GraphQLClient.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export interface GraphQLClientOptions {
   monitoring?: Monitoring;
   getAccessToken?: GetAccessTokenFn;
   defaultFetchPolicy?: FetchPolicy;
+  overriddenHttpLink?: ApolloLink;
```
Comment (Smtih) [inline src/GraphQLClient.ts:26]: NIT: I think you can simplify this to just `httpLink`
Pretty common to provide config variables that will have a default value when not provided. Don't think we need to specify override.

**Review by Smtih — DISMISSED** [review]
LGTM

File: src/GraphQLClient.ts (hunk 1 above)
Comment (hungnguyen-ailo) [reply src/GraphQLClient.ts:26]: Yeah makes sense, updating

Comment (hungnguyen-ailo) [reply src/GraphQLClient.ts]: Updated in https://github.com/ailohq/graphql-client/pull/16/commits/fc00ab2b47562ada367af93549c1d94b3ad03860, thank you 🙇

**Review by Smtih — APPROVED** [review]

**Pull request merged**